apis/r/R/pad_matrix.R

```diff
@@ -25,12 +22,8 @@ pad_matrix <- function(x, ...) {
 #' row/column names
 #'
 #' @importFrom Matrix sparseMatrix
-#'
-#' @noRd
-#'
 #' @method pad_matrix default
-#' @export
```

Comment:
The `@export` tags are (IIRC) still needed for the method definitions to enable S3 method dispatch (`@export` on a function vs `@export` on a method are different)

Reply:
(looking at the diff for `NAMESPACE` shows that the method definitions are no longer registered)

Reply:
I have @export on a ton of methods in tiledb-r too.  Plus occasional exportClass.  

I think roxygen2 behavior changed here over the years.

(Usual disclaimer:  roxygen2 is something "we" (as in R users) do, it is not normative or even mentioned really in WRE.  Only the final NAMESPACE and man/ matter.)

Reply:
(Disclaimer:  It is correct _if_ one wants the method public. )

Reply:
Not by default. You have to `@rdname` the method into the file for the generic (which is why only `\method{pad_matrix}{matrix}` shows up, but not `\method{pad_matrix}{default}`; I had these separated for reasons that I cannot remember)

Reply:
@mojaveazure are you suggesting S3 dispatch for pad_matrix() will not work if it's not publicly exported?

Reply:
I made a quick toy example to test and it doesn't appear to be a problem. Check it out and let me know if I've missed anything: https://github.com/aaronwolen/reverse

Reply:
Using @aaronwolen's example package, it seems as though S3 dispatch works as long as usage is within-namespace. The moment we call the generic from outside of the namespace, we lose dispatch. I think this is acceptable behavior for `pad_matrix()`

![image](https://github.com/single-cell-data/TileDB-SOMA/assets/11035068/9910762d-49c2-491a-9641-43325e2ae72c)

Reply:
Isn't this a non-issue?  Once we decide to not export it, we pretty much say it won't (or may not) work outside the package -- and we care mostly about the use inside the package. So ... onwards!